### ISSUE TITLE
[a11y]: ghost button lacks sufficient text contrast on focus (and possibly without focus too)

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#10177]
### Package

carbon-components

### Browser

Chrome

### Operating System

MacOS

### Package version

Carbondesignsystem.com

### Automated testing tool and ruleset

visual confirmation plus contrast tool

### Assistive technology

n/a

### Description

The text colour of the Ghost button fades significantly on keyboard focus. There is no reason this should happen; all that needs to happen is for the blue focus indicator to be added around the existing text.

Incidentally, I believe that the text of the Ghost button may be below text contrast minimums _before_ it receives focus. I measure it as 3.9:1. It needs to be 4.5:1. This _could_ be a sampling issue.
This can be achieved by using the same blue for text as the tertiary button

### WCAG 2.1 Violation

1.4.3 Contrast (Minimum)

### CodeSandbox example

https://www.carbondesignsystem.com/components/button/code/

### Steps to reproduce

1. Go to the Code tab of the Button documentation in Carbon
1. Select the ghost button from the Kind radio button group
1. Use a contrast tool on the text colour of the ghost button to see if it achieves 4.5:1 against the background
1. Click on the Variant selector to move focus near the Ghost button, then press Tab until the ghost button receives focus
2. The text has clearly faded.

![image](https://user-images.githubusercontent.com/14298245/143602835-04fd60ec-122e-4dbe-a53d-b17ac7d74e7a.png)
^ Ghost button without focus

![image](https://user-images.githubusercontent.com/14298245/143602783-1fa0824e-4549-4c2f-b4ae-424e0e94a871.png)
^ ghost button with focus

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by brunnerh** [comment]
The color does not change for me 🤔 (Windows 10, in Chrome and Firefox).

![image](https://user-images.githubusercontent.com/834235/143779467-cdf8545c-312e-4465-b2ec-feba490d85ce.png)

**Comment by mbgower** [comment]
Hmm, I verified it doesn't change with me under the same circumstances. let me get another Mac user to try...

**Comment by marcjohlic** [comment]
Verified that I'm not seeing the fading behavior on focus - and the text contrast appears to be sufficient at 5.0:1  (on MacOS BigSur and Firefox)

![image](https://user-images.githubusercontent.com/6586306/145051645-004df75d-812e-4e6d-8605-266512b92bc9.png)


![image](https://user-images.githubusercontent.com/6586306/145051602-07006b15-d28b-410e-b797-ad282be9ab8e.png)

#0f62fe

**Comment by mbgower** [comment]
I tried upgrading to Monterey to see if this resolved. No difference. Based on Marc's findings, I think hunted around and realized it was a result of some settings I had in Accessibility.
@joshblack I had both Increase Contrast and Reduce Transparency selected. Once I cleared those, the issue went away.

![image](https://user-images.githubusercontent.com/14298245/145053117-a3349556-32b1-4528-a3e7-4fcfae5f89af.png)

**Comment by grahamharper** [comment]
@mbgower does ghost button fail [SC 1.4.1 Use of Color](https://www.w3.org/TR/2016/NOTE-UNDERSTANDING-WCAG20-20161007/visual-audio-contrast-without-color.html)?

Or are we satisfied that we are using a [sufficient technique](https://www.w3.org/TR/2016/NOTE-UNDERSTANDING-WCAG20-20161007/visual-audio-contrast-without-color.html#visual-audio-contrast-without-color-techniques-head)?

Or does it require another visual style change on hover/focus as per [G183](https://www.w3.org/TR/WCAG20-TECHS/G183.html)? e.g. an underline, border, font change?

**Comment by mbgower** [comment]
@grahamharper this issue was about the text in a ghost button not having sufficient text contrast.

If you have a related but different question, I think it makes sense to open a new issue specifically on ghost buttons and use of color, rather than tack it onto a closed issue. That okay?
